src/app/shared/services/filter-data.service.ts

```diff
@@ -14,13 +12,55 @@ export class FilterDataService {
     'goals-pending': [],
     'task-summary': [],
     'member-summary': [],
-    };
+  };
+  coloumData = {};
+  constructor() { }
 
+  setColumnData(retroId, tableKey, data) {
```

Comment:
1. Please specify types of the parameters.
2. This name looks better.
```suggestion
  saveColumnState(retroID: string, tableKey: string, columnState: any) {
```

Reply:
done